proxy/src/http/avatar.rs

```diff
@@ -0,0 +1,99 @@
+//! Endpoints for Avatar.
+
+use warp::document::{self, ToDocumentedType};
+use warp::{path, Filter, Rejection, Reply};
+
+use crate::avatar;
+
+/// `GET /avatars/<usage>/<id>`
```

Comment:
It seems usage might be better suited as a query param, with the default being `any`. In turn we would return a `400`, if the supplied value is not supported.

Reply:
I moved `usage` into the query.
For the `400`, do I have to create a new error in `src/error.rs`?